sdk/core/Azure.Core/tests/public/ModelSerializationTests/AvailabilitySetDataTestsWithVMs.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Azure.Core.Serialization;
+using Azure.Core.Tests.Common;
+using Azure.Core.Tests.ResourceManager.Compute;
+using NUnit.Framework;
+
+namespace Azure.Core.Tests.Public.ModelSerializationTests
+{
+    internal class AvailabilitySetDataTestsWithVMs : ModelJsonTests<AvailabilitySetData>
+    {
+        protected override string WirePayload => File.ReadAllText(TestData.GetLocation("AvailabilitySetDataWithVMs.json")).TrimEnd();
+
+        protected override string JsonPayload => WirePayload;
+
+        protected override Func<AvailabilitySetData, RequestContent> ToRequestContent => model => model;
+
+        protected override Func<Response, AvailabilitySetData> FromResponse => response => (AvailabilitySetData)response;
+
+        protected override string GetExpectedResult(ModelSerializerFormat format)
+        {
+            var expectedSerializedString = "{";
```

Comment:
lets refactor this so that a test data model has a wire and json format file along side it

TestData
 - Model.json
 - ModelWire.json
 - ModelJson.json

This way we can validate very large objects as well without walls of text